ophyd/status.py

```diff
@@ -57,15 +57,19 @@ def __init__(self, *, timeout=None, settle_time=None, done=False,
             return
 
         if self.timeout is not None and self.timeout > 0.0:
-            thread = threading.Thread(target=self._timeout_thread, daemon=True)
+            thread = threading.Thread(target=self._wait_and_cleanup,
+                                      daemon=True)
             self._timeout_thread = thread
             self._timeout_thread.start()
 
-    def _timeout_thread(self):
+    def _wait_and_cleanup(self):
```

Comment:
Note: The original name of this method collided with the name of the attribute where the thread was stash. It doesn't matter because the method is only called once, and it might even have been intentional, butI found it confuses so I'm leaving the change in.